sec_certs_page/common/bindings.py

```diff
@@ -0,0 +1,510 @@
+                for binding in bindings:
+                    binding.update_headers_if_match(
+                        file_name, json_obj["data"])
+
+    click.echo(f"Downloading {len(tasks)} metadata files")
+    await asyncio.gather(*tasks)
+
+
+def get_instance_paths() -> Tuple[str, str, str]:
+    """ Get instance paths for bindings, headers and metadata files
+
+    Returns:
+        Tuple[str, str, str]: The three bindings, headers and metadata files directories paths
+    """
+    instance_path = current_app.instance_path
+    return (
+        os.path.join(instance_path, "bindings"),
+        os.path.join(instance_path, "headers"),
+        os.path.join(instance_path, "metadata")
+    )
+
+
+def process_and_download_all(
+        download_bindings: bool = True,
+        download_headers: bool = True,
+        download_metadata: bool = True,
+        verbose: bool = False) -> list[Binding]:
+    """ Encapsulates the function calls to download all bindings, process them and to process all header files
+
+    Args:
+        download_bindings (bool, optional): Whether to download binding files. Defaults to True.
+        download_headers (bool, optional): Whether to download header files. Defaults to True.
+        download_metadata (bool, optional): Whether to download metadata files. Defaults to True.
+        verbose (bool, optional): Whether to click.echo out extra information. Defaults to False.
+
+    Returns:
+        list[Binding]: _description_
+    """
+    bindings_path, headers_path, metadata_path = get_instance_paths()
+    binding_urls = current_app.config["BINDING_URLS"]
+    if download_bindings:
+        download_all_bindings_sync(
+            binding_urls, bindings_path, verbose=verbose)
+    bindings = asyncio.run(process_binding_files(
+        bindings_path, headers_path, download_headers, verbose=verbose))
+    asyncio.run(
+        process_header_files(
+            headers_path,
+            metadata_path,
+            bindings,
+            download_metadata,
+            verbose=verbose))
+    return bindings
+
+
+def purge_headers_data(collection: Collection[Document]) -> None:
+    """ Purges all existing headers_data from the MongoDB collection
+
+    Args:
+        collection (Collection[Document]): MongoDB collection to purge headers from
+    """
+    result = collection.delete_many({})
+    print(f"Deleted {result.deleted_count} metadata bindings")
+
+
+def update_one(
+        collection: Collection[Document],
+        binding: Binding,
+        verbose: bool = False) -> None:
+    """ Updates a record in mongoDB with new header_data
+
+    Args:
+        collection (Collection[Document]): MongoDB collection to update the records in
+        binding (Binding): An object representing the binding - cert_id and header_data
+        verbose (bool, optional): Whether to click.echo out extra information. Defaults to False.
+    """
+    cert_id = binding["cert_id"] if "NIST" not in binding["cert_id"] else int(
```

Comment:
This is nasty. The binding should just have another attribute that says whether it is CC or FIPS. Nothing like "NIST-" prefixed cert IDs exists anywhere.